error: generic arguments in macro path
  --> $DIR/macro-ty-params.rs:20:8
   |
LL |     foo::<T>!(); //~ ERROR generic arguments in macro path
   |        ^^^^^

error: generic arguments in macro path
  --> $DIR/macro-ty-params.rs:21:8
   |
LL |     foo::<>!(); //~ ERROR generic arguments in macro path
   |        ^^^^

error: unexpected generic arguments in path
  --> $DIR/macro-ty-params.rs:22:8
   |
LL |     m!(Default<>); //~ ERROR generic arguments in macro path
   |        ^^^^^^^^^

error: generic arguments in macro path
  --> $DIR/macro-ty-params.rs:22:15
   |
LL |     m!(Default<>); //~ ERROR generic arguments in macro path
   |               ^^

error: aborting due to 4 previous errors

